### PR TITLE
add new exchanger based on exchanger package

### PR DESCRIPTION
**Pull request by Ellrion**
Swap it's wrapper for `florianv/exchanger`. For some projects swap may be redundant

### CODE REVIEW

**Comment by sagikazarmark** [comment]
Thanks for the contribution!

Can you please clarify what you mean by redundant?

When we implemented the new (v3) Swap we followed @florianv's suggestion do implement it instead of exchanger.

File: src/Exchange/ExchangerExchange.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * Provides a way to get exchange rate from a third-party source and return a currency pair.
+ *
+ * @author Maksim (Ellrion) Platonov <ellrion11@gmail.com>
+ */
+final class ExchangerExchange implements Exchange
+{
+    /**
+     * @var \Exchanger\Contract\ExchangeRateProvider
```
Comment (sagikazarmark) [inline src/Exchange/ExchangerExchange.php]: Please don't use the FQCN.

File: src/Exchange/ExchangerExchange.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+     */
+    private $exchanger;
+
+    /**
+     * @param \Exchanger\Contract\ExchangeRateProvider $exchanger
```
Comment (sagikazarmark) [inline src/Exchange/ExchangerExchange.php]: Please don't use the FQCN.

**Comment by Ellrion** [comment]
> Can you please clarify what you mean by redundant?

I use `florianv/exchanger` (not `florianv/swap`). Becouse swap it wrapper for exchanger. I use many implementation of  `Exchanger\Contract\ExchangeRateProvider` and when i want to use Money\Exchange 
I have code like this

```php
/**
 * @var Exchanger\Contract\ExchangeRateProvider $service
 */
$exchange =  new SwapExchange(new Swap(new Exchanger($service)));
$converter = new Converter($this->app['money.currencies'], $exchange)
```

but better

```php
/**
 * @var Exchanger\Contract\ExchangeRateProvider $service
 */
$exchange =  new ExchangerExchange($service));
$converter = new Converter($this->app['money.currencies'], $exchange)
```
 And there will be no dependence on `florianv/swap`

Please excuse my bad English

**Comment by florianv** [comment]
Hey,

> When we implemented the new (v3) Swap we followed @florianv's suggestion do implement it instead of exchanger.

Indeed, I didn't expect people would use `Exchanger` directly as framework integrations provide the `Swap` service and don't expose `Exchanger`.

To me `Swap` is more convenient but from some issues and suggestions I got, it seems that some people use `Exchanger` directly, maybe without a framework.

This implementation looks good to me. Maybe we could merge both but I'm not sure which name to use for the implementation

**Comment by Ellrion** [comment]
@frederikbosch Whot about it?

**Comment by frederikbosch** [comment]
@Ellrion Sorry, I leave this up to @sagikazarmark. My knowledge of the exchange part of the library is limited. I cannot give a comprehensive review of this feature.

**Comment by sagikazarmark** [comment]
@Ellrion sorry for the huge delay, this will be part of 3.2.0 release (no ETA yet, but hopefully soon).

**Comment by sagikazarmark** [comment]
@Ellrion can you please update your PR to use the latest stable version?

**Comment by sagikazarmark** [comment]
@Ellrion thanks a lot for your work. I made the requested changes myself.

**Pull request closed without merging**